tests/components/mcp_server/test_http.py

```diff
@@ -0,0 +1,345 @@
+    assert response.status == HTTPStatus.NOT_FOUND
+    response_data = await response.text()
+    assert "Model Context Protocol server is not configured" in response_data
+
+
+async def test_http_messages_no_config_entry(
+    hass: HomeAssistant,
+    setup_integration: None,
+    config_entry: MockConfigEntry,
+    hass_client: ClientSessionGenerator,
+) -> None:
+    """Test the message endpoint will fail if the config entry is unloaded."""
+
+    client = await hass_client()
+
+    # Start an SSE session
+    response = await client.get(SSE_API)
+    assert response.status == HTTPStatus.OK
+    it = aiter(response.content)
+    line = (await anext(it)).decode()
+    assert line == EXPECTED_ENDPOINT_EVENT
+    line = (await anext(it)).decode()
+    assert line.startswith(EXPECTED_ENDPOINT_DATA)
+    endpoint_url = line[6:].strip()
+
+    # Invalidate the session by unloading the config entry
+    await hass.config_entries.async_unload(config_entry.entry_id)
+    assert config_entry.state is ConfigEntryState.NOT_LOADED
+
+    # Reload the config entry and ensure the session is not found
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    response = await client.post(endpoint_url, json=INITIALIZE_MESSAGE)
+    assert response.status == HTTPStatus.NOT_FOUND
+    response_data = await response.text()
+    assert "Could not find session ID" in response_data
+
+
+async def test_http_requires_authentication(
+    hass: HomeAssistant,
+    setup_integration: None,
+    hass_client_no_auth: ClientSessionGenerator,
+) -> None:
+    """Test the SSE endpoint requires authentication."""
+
+    client = await hass_client_no_auth()
+
+    response = await client.get(SSE_API)
+    assert response.status == HTTPStatus.UNAUTHORIZED
+
+    response = await client.post(MESSAGES_API.format(session_id="session-id"))
+    assert response.status == HTTPStatus.UNAUTHORIZED
+
+
+@pytest.fixture
+async def mcp_sse_url(hass_client: ClientSessionGenerator) -> str:
+    """Fixture to get the MCP integration SSE URL."""
+    client = await hass_client()
+    return str(client.make_url(SSE_API))
+
+
+@asynccontextmanager
+async def mcp_session(
+    mcp_sse_url: str,
+    hass_supervisor_access_token: str,
+) -> AsyncGenerator[mcp.client.session.ClientSession]:
+    """Create an MCP session."""
+
+    headers = {"Authorization": f"Bearer {hass_supervisor_access_token}"}
+
+    async with (
+        mcp.client.sse.sse_client(mcp_sse_url, headers=headers) as streams,
+        mcp.client.session.ClientSession(*streams) as session,
+    ):
+        await session.initialize()
+        yield session
+
+
+async def test_mcp_tools_list(
+    hass: HomeAssistant,
+    setup_integration: None,
+    mcp_sse_url: str,
+    hass_supervisor_access_token: str,
+) -> None:
+    """Test the tools list endpoint."""
+
+    async with mcp_session(mcp_sse_url, hass_supervisor_access_token) as session:
```

Comment:
For reviewer: This is using the MCP SDK client library for testing the MCP and SSE protocol for the integration end to end. This is not typical for a Home Assistant test, but since this is adding a new protocol it seemed to make sense.  If desired, we can make the tests lower level.

Reply:
I like this, because it guarantees that our implementation works instead of trying to keep our lower level tests in sync with the spec.